crates/ruff_python_parser/src/lexer.rs

```diff
@@ -2045,9 +2045,12 @@ f"{(lambda x:{x})}"
     #[test]
     fn test_fstring_error() {
         use FStringErrorType::{
-            UnclosedLbrace, UnterminatedString, UnterminatedTripleQuotedString,
+            SingleRbrace, UnclosedLbrace, UnterminatedString, UnterminatedTripleQuotedString,
         };
 
+        assert_eq!(lex_fstring_error(r#"f"{a:b}}""#), SingleRbrace);
+        assert_eq!(lex_fstring_error(r#"f"}""#), SingleRbrace);
```

Comment:
This is a good find. The intention was not to check for infinite loop. The test function stops when it encounters an error but the lexer goes into an infinite loop when it's otherwise used.

Reply:
Also, note that `cargo run --bin ruff` won't work right now as the linter changes w.r.t. the new f-string tokens are yet to be implemented.